crates/forge/tests/integration/start_prank.rs

```diff
@@ -177,3 +177,69 @@ fn start_prank_in_constructor_test() {
 
     assert_passed!(result);
 }
+
+#[test]
+fn start_prank_with_proxy() {
+    let test = test_case!(
+        indoc!(
+            r#"
+            use result::ResultTrait;
+            use array::ArrayTrait;
+            use option::OptionTrait;
+            use traits::TryInto;
+            use traits::Into;
+            use starknet::ContractAddress;
+            use starknet::Felt252TryIntoContractAddress;
+            use cheatcodes::PreparedContract;
+            use forge_print::PrintTrait;
+
+            #[starknet::interface]
+            trait IPrankCheckerProxy<TContractState> {
+                fn get_prank_checkers_caller_address(ref self: TContractState, address: ContractAddress) -> felt252;
+            }
+
+            #[test]
+            fn test_prank_simple() {
+                let class_hash = declare('PrankChecker');
+                let prepared = PreparedContract { class_hash: class_hash, constructor_calldata: @ArrayTrait::new() };
+                let prank_checker_contract_address = deploy(prepared).unwrap();
+                let prank_checker_contract_address: ContractAddress = prank_checker_contract_address.try_into().unwrap();
+                let contract_address: ContractAddress = 234.try_into().unwrap();
+                start_prank(contract_address, prank_checker_contract_address);
+                
+                let class_hash = declare('PrankCheckerProxy');
+                let prepared = PreparedContract { class_hash: class_hash, constructor_calldata: @ArrayTrait::new() };
+                let proxy_contract_address = deploy(prepared).unwrap();
+                let proxy_contract_address: ContractAddress = proxy_contract_address.try_into().unwrap();
+                let proxy_dispatcher = IPrankCheckerProxyDispatcher { contract_address: proxy_contract_address };
+
+                let caller_address = proxy_dispatcher.get_prank_checkers_caller_address(prank_checker_contract_address);
+                assert(caller_address == 234, caller_address);
+            }
```

Comment:
We should make sure the address changed/block number to the previous one (applies to all prank and roll) 

```
let a = call get_block_number

start_roll 234
let x = call get_block_number
assert x == 234

stop roll
let b = call_get_block_number

assert a == b
```

Reply:
https://github.com/foundry-rs/starknet-foundry/issues/341

Reply:
#303 added some of it